profiler/src/test/java/com/splunk/opentelemetry/profiler/context/StackDescriptorLineParserTest.java

```diff
@@ -28,79 +30,120 @@
     assertEquals(CANT_PARSE_THREAD_ID, result);
   }
 
   @Test
   void testPurposefullyDeviantInput() {
-    StackDescriptorLineParser parser = new StackDescriptorLineParser();
-    long result = parser.parseThreadId(".#31 daemon x");
+    long result = parseAndAssertWithAffixes(".#31 daemon x");
     assertEquals(CANT_PARSE_THREAD_ID, result);
   }
 
   @Test
   void testQuoteAtEnd() {
-    StackDescriptorLineParser parser = new StackDescriptorLineParser();
     long result =
-        parser.parseThreadId(
+        parseAndAssertWithAffixes(
             ".#31 daemon prio=8 os_prio=31 cpu=0.41ms elapsed=50.48s tid=0x00007fb517030000\"");
     assertEquals(CANT_PARSE_THREAD_ID, result);
   }
 
   @Test
   void testEntireLineQuoted() {
-    StackDescriptorLineParser parser = new StackDescriptorLineParser();
     long result =
-        parser.parseThreadId(
+        parseAndAssertWithAffixes(
             "\".#31 daemon prio=8 os_prio=31 cpu=0.41ms elapsed=50.48s tid=0x00007fb517030000\"");
     assertEquals(CANT_PARSE_THREAD_ID, result);
   }
 
   @Test
   void testFallOffTheEnd() {
-    StackDescriptorLineParser parser = new StackDescriptorLineParser();
     long result =
-        parser.parseThreadId(
+        parseAndAssertWithAffixes(
             "\".#31 daemon prio=8 os_prio=31 cpu=0.41ms elapsed=50.48s tid=0x00007fb517030000\" ");
     assertEquals(CANT_PARSE_THREAD_ID, result);
   }
 
   @Test
   void testFallOffTheEndWithHash() {
-    StackDescriptorLineParser parser = new StackDescriptorLineParser();
-    long result = parser.parseThreadId("\"test\" #");
+    long result = parseAndAssertWithAffixes("\"test\" #");
     assertEquals(CANT_PARSE_THREAD_ID, result);
   }
 
   @Test
   void testSecondSpaceMissing() {
-    StackDescriptorLineParser parser = new StackDescriptorLineParser();
-    long result = parser.parseThreadId("\"test\" #33");
+    long result = parseAndAssertWithAffixes("\"test\" #33");
     assertEquals(CANT_PARSE_THREAD_ID, result);
   }
 
   @Test
   void testNotNumeric() {
-    StackDescriptorLineParser parser = new StackDescriptorLineParser();
-    long result = parser.parseThreadId("\"test\" #zoinks ");
+    long result = parseAndAssertWithAffixes("\"test\" #zoinks ");
     assertEquals(CANT_PARSE_THREAD_ID, result);
   }
+
+  @Test
+  void testEmptyString() {
+    long result = parseAndAssertWithAffixes("");
+    assertEquals(CANT_PARSE_THREAD_ID, result);
+  }
+
+  long parseAndAssertWithAffixes(String descriptorLine) {
+    StackDescriptorLineParser parser = new StackDescriptorLineParser();
+    ThreadDumpRegion region = new ThreadDumpRegion(descriptorLine, 0, descriptorLine.length());
+    long result = parser.parseThreadId(region);
+
+    Stream.of(DISRUPTIVE_AFFIXES)
+        .forEach(
+            it -> {
+              assertEqualsWithAffixes(result, parser, descriptorLine, "", it);
+              assertEqualsWithAffixes(result, parser, descriptorLine, it, "");
+            });
```

Comment:
This could be improved later maybe by leveraging junit dynamic tests?

Reply:
Possibly, but not going to look into it for the current PR.